apps/web/app/api/passport-oracle/addStrategy/route.ts

```diff
@@ -0,0 +1,144 @@
+// pages/api/addStrategy.ts
+
+import { NextResponse } from "next/server";
+import {
+  createPublicClient,
+  http,
+  createWalletClient,
+  custom,
+  Address,
+  Chain,
+} from "viem";
+import { privateKeyToAccount } from "viem/accounts";
+import { localhost, arbitrumSepolia, sepolia, arbitrum } from "viem/chains";
+import { getConfigByChain } from "@/constants/contracts";
+import {
+  passportScorerABI,
+  registryCommunityABI,
+  cvStrategyABI,
+} from "@/src/generated";
+
+const LIST_MANAGER_PRIVATE_KEY = process.env.LIST_MANAGER_PRIVATE_KEY;
+
+const CHAIN = process.env.CHAIN_ID ? parseInt(process.env.CHAIN_ID) : 1337;
+const LOCAL_RPC = "http://127.0.0.1:8545";
+
+const RPC_URL = getConfigByChain(CHAIN)?.rpcUrl ?? LOCAL_RPC;
+
+const PASSPORT_SCORER_ADDRESS = getConfigByChain(CHAIN)
+  ?.passportScorer as Address;
+
+function getViemChain(chain: number): Chain {
```

Comment:
Can we have it extracted somewhere rather than duplicating it?

Reply:
yess yes you were quicker than me , i pushed some stuffs and then proceed to do the upgredable stuff and fix the vulnerabiltity that lucho found, but rest asure that will push what you said